File: POO JAVA/Aula11/src/aula11/Professor.java
package aula11;

public class Professor extends Pessoa{

    //Atribustos da Classe
    private String especialidade;
    private float salario;
    
    //Metodos Acessores e Modificadores
    public String getEspecialidade() {
        return especialidade;
    }
    public void setEspecialidade(String especialidade) {
        this.especialidade = especialidade;
    }
    public float getSalario() {
        return salario;
    }
    public void setSalario(float salario) {
        this.salario = salario;
    }
    
    //Metodos da Classe
    public void receberSalario(){
        System.out.println(this.nome+" professor especialista em "+this.getEspecialidade()+" recebendo o salario mensal.");
    }
}

File: POO PHP/Aula11/Professor.php
<?php

require_once 'Pessoa.php';

class Professor extends Pessoa{

    //Atributos da Classe
    private $especialização;
    private $salario;
    
    //Metodos Acessores e Modificadores
    public function getEspecialização() {
        return $this->especialização;
    }
    public function getSalario() {
        return $this->salario;
    }
    public function setEspecialização($especialização): void {
        $this->especialização = $especialização;
    }
    public function setSalario($salario): void {
        $this->salario = $salario;
    }

    //Metodo da Classe
    public function receberSalario($param) {
        $this->salario = $param;
    }
}
